pico/tools/src/main/java/io/helidon/pico/tools/CodeGenFiler.java

```diff
@@ -64,13 +63,13 @@ public class CodeGenFiler {
     private static final boolean FORCE_MODULE_INFO_PICO_INTO_SCRATCH_DIR = true;
     private static final boolean FILER_WRITE_ONCE_PER_TYPE = true;
     private static final Set<TypeName> FILER_TYPES_FILED = new LinkedHashSet<>();
-    private static boolean filerWriteIsDisabled;
+    private static boolean filerWriteEnabled = true;
```

Comment:
static field that is not volatile - this may not work the way you want (as it does not seem to be guarded by any lock)

Reply:
seems to be working fine right now, idk

Reply:
To rephrase it: if there is a possibility that the field will be used from more than one thread, it MUST be either volatile or an `AtomicBoolean`.

Reply:
pushed to use volatile